engine/consensus/compliance/core.go

```diff
@@ -230,22 +235,52 @@ func (c *Core) OnBlockProposal(originID flow.Identifier, proposal *messages.Bloc
 func (c *Core) processBlockAndDescendants(proposal *messages.BlockProposal) error {
 	blockID := proposal.Header.ID()
 
+	// retrieve the parent block
+	//  - once we reach this point, the parent block must have been validated
+	//    and inserted to the protocol state
+	parent, err := c.headers.ByBlockID(proposal.Header.ParentID)
```

Comment:
We are querying this parent second time here. Consider using the parent queried [the first time here](https://github.com/onflow/flow-go/pull/3294/files#diff-91e83ea2456f635c82eea90128d3f2820974b95947215b9a75da64c0f513b5e5R194)

Reply:
Nice catch, thanks https://github.com/onflow/flow-go/pull/3294/commits/64ca513cd99b5fc0216fe9a7747d266fd1d1baec